src/Features/CSharp/Portable/SignatureHelp/PrimaryConstructorBaseTypeSignatureHelpProvider.cs

```diff
@@ -0,0 +1,153 @@
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SignatureHelp;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp
+{
+    /// <summary>
+    /// Implements SignatureHelp and ParameterInfo for <see cref="PrimaryConstructorBaseTypeSyntax"/>
+    /// such as 'record Student(int Id) : Person($$"first", "last");`.
+    /// </summary>
+    [ExportSignatureHelpProvider("PrimaryConstructorBaseTypeSignatureHelpProvider", LanguageNames.CSharp), Shared]
+    internal partial class PrimaryConstructorBaseTypeSignatureHelpProvider : AbstractCSharpSignatureHelpProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public PrimaryConstructorBaseTypeSignatureHelpProvider()
+        {
+        }
+
+        public override bool IsTriggerCharacter(char ch)
+            => ch == '(' || ch == ',';
+
+        public override bool IsRetriggerCharacter(char ch)
+            => ch == ')';
+
+        private bool TryGetBaseTypeSyntax(SyntaxNode root, int position, ISyntaxFactsService syntaxFacts, SignatureHelpTriggerReason triggerReason, CancellationToken cancellationToken, out PrimaryConstructorBaseTypeSyntax expression)
+        {
+            if (!CommonSignatureHelpUtilities.TryGetSyntax(root, position, syntaxFacts, triggerReason, IsTriggerToken, IsArgumentListToken, cancellationToken, out expression))
+            {
+                return false;
+            }
+
+            return expression.ArgumentList != null;
+
+            static bool IsArgumentListToken(PrimaryConstructorBaseTypeSyntax expression, SyntaxToken token)
+            {
+                return expression.ArgumentList != null &&
+                    expression.ArgumentList.Span.Contains(token.SpanStart) &&
+                    token != expression.ArgumentList.CloseParenToken;
+            }
+        }
+
+        private bool IsTriggerToken(SyntaxToken token)
+            => SignatureHelpUtilities.IsTriggerParenOrComma<PrimaryConstructorBaseTypeSyntax>(token, IsTriggerCharacter);
+
+        protected override async Task<SignatureHelpItems> GetItemsWorkerAsync(Document document, int position, SignatureHelpTriggerInfo triggerInfo, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (!TryGetBaseTypeSyntax(root, position, document.GetLanguageService<ISyntaxFactsService>(), triggerInfo.TriggerReason, cancellationToken, out var baseTypeSyntax))
+            {
+                return null;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var within = (INamedTypeSymbol)semanticModel.GetDeclaredSymbol(baseTypeSyntax.Parent.Parent, cancellationToken);
```

Comment:
would it be slightly cleaner/clearer to use GetEnclosingNamedType?  since you say 'within', it seems like that would be a clearer way of asking the question.  thoughts?

Reply:
Sure. Done :-)